auth/overview_snippets.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package authsnippets contains Google Cloud authentication snippets.
+package authsnippets
+
+import (
+	"context"
+	"fmt"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+	"google.golang.org/api/pubsub/v1"
+	"log"
+	"os"
+)
+
+// [START auth_overview_service_account]
+
+// Use service account to authenticate.
```

Comment:
Please start comments with the thing being described.

For example:

```
// serviceAccount shows how to authenticate with a service account set by the GOOGLE_APPLICATION_CREDENTIALS environment variable.
```

https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences